tuf/tuf_test.go

```diff
@@ -166,6 +166,144 @@ func TestUpdateDelegations(t *testing.T) {
 	require.False(t, ok, "no empty targets file should be created for deepest delegation")
 }
 
+func TestPurgeDelegationsKeyFromTop(t *testing.T) {
```

Comment:
in this test or a new one, can we test what happens if we:

 - try to purge a key ID that isn't signed into any delegation
 - remove the targets private key from our keystore, and try to purge a valid key ID from delegations